tests/test_configuration.py

```diff
@@ -66,3 +66,84 @@ def test_argument_parsing(self, mock_sys):
+        configuration.setup()
+        self.assertListEqual(configuration.scan_arguments, [])
+
+        test_dict = {"tool": "some_property"}
+        configuration._read_toml_file = Mock(return_value=test_dict)
+        configuration.setup()
+        self.assertListEqual(configuration.scan_arguments, [])
+
+        test_dict = {"tool": {"sonar": "some_property"}}
+        configuration._read_toml_file = Mock(return_value=test_dict)
+        configuration.setup()
+        self.assertListEqual(configuration.scan_arguments, [])
+
+    @patch("py_sonar_scanner.configuration.sys")
+    def test_dict_with_valid_values(self, mock_sys):
+        configuration = Configuration()
+        mock_sys.argv = ["path/to/scanner/py-sonar-scanner"]
+
+        test_dict = {"tool": {"sonar": {"property1": "value1"}}}
+        configuration._read_toml_file = Mock(return_value=test_dict)
+        configuration.setup()
+        self.assertListEqual(configuration.scan_arguments, ["-Dsonar.property1=value1"])
+
+        test_dict = {"tool": {"sonar": {"property1": "value1", "property2": "value2"}}}
+        configuration._read_toml_file = Mock(return_value=test_dict)
+        configuration.setup()
+        self.assertListEqual(configuration.scan_arguments, ["-Dsonar.property1=value1", "-Dsonar.property2=value2"])
+
+        test_dict = {"tool": {"sonar": {"property_class": {"property1": "value1"}}}}
+        configuration._read_toml_file = Mock(return_value=test_dict)
+        configuration.setup()
+        self.assertListEqual(configuration.scan_arguments, ["-Dsonar.property_class.property1=value1"])
+
+        test_dict = {"tool": {"sonar": {"property1": "value1", "property_class": {"sub_property": "sub_value"}}}}
+        configuration._read_toml_file = Mock(return_value=test_dict)
+        configuration.setup()
+        self.assertListEqual(
+            configuration.scan_arguments, ["-Dsonar.property1=value1", "-Dsonar.property_class.sub_property=sub_value"]
+        )
+
+    @patch("py_sonar_scanner.configuration.sys")
+    def test_toml_with_valid_values(self, mock_sys):
+        configuration = Configuration()
+        mock_sys.argv = ["path/to/scanner/py-sonar-scanner", "-Dtoml.path=tests/resources/test_toml_file.toml"]
+        configuration.setup()
+        self.assertListEqual(
+            configuration.scan_arguments,
+            [
+                "-Dtoml.path=tests/resources/test_toml_file.toml",
+                "-Dsonar.property1=value1",
+                "-Dsonar.property2=value2",
+                "-Dsonar.property_class.property1=value1",
+            ],
+        )
+
+    @patch("py_sonar_scanner.configuration.ApplicationLogger.get_logger")
+    @patch("builtins.open")
+    @patch("py_sonar_scanner.configuration.sys")
+    def test_error_while_reading_toml_file(self, mock_sys, mock_open, mock_logger):
+        configuration = Configuration()
+
+        mock_sys.argv = ["path/to/scanner/py-sonar-scanner"]
```

Comment:
Similar comment about file paths here.

This test currently basically works accidentally when ran through the `pytest` command at the root of the repository and fails when run from the gutter icon.

Basically, when run from `pytest`, it picks up and tries to open the actual `pyproject.toml` of the project root instead of one from our tests, because the `-Dtoml_file.toml` is not set. When ran from the gutter icon, no `pyproject.toml` file is present and we just don't try to open it.

I suggest to define the `toml_path` property in a similar fashion to the other tests so that we always find a suitable test `toml` file.